packages/noya-state/src/reducers/themeReducer.ts

```diff
@@ -505,6 +529,19 @@ export function themeReducer(
         sharedSwatches.objects = filterSwatches;
 
         draft.sketch.document.sharedSwatches = sharedSwatches;
+
+        layerIndexPathsWithSwatchId.forEach((layerPath) =>
+          accessPageLayers(
+            draft,
+            layerPath.pageIndex,
+            layerPath.indexPaths,
+          ).forEach((layer) => {
+            layer.style?.fills?.forEach((f) => {
+              if (f.color.swatchID && ids.includes(f.color.swatchID))
```

Comment:
Very minor but we already checked this condition above, so I think it should be safe to just `delete` the swatch without the condition. Anything in `layerIndexPathsWithSwatchId` should have the correct swatchID

Reply:
The layer can have multiple fills

Reply:
Oh yeah, good point. Ok, keep it the way it is